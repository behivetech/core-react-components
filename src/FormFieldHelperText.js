// Vendor Libs
import React, {Component, createRef} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {MDCTextFieldHelperText} from '@material/textfield/helper-text';

export default class FormFieldHelperText extends Component {
    constructor(props) {
        super(props);

        this.textFormFieldHelperTextRef = createRef();
    }

    componentDidMount() {
        new MDCTextFieldHelperText(this.textFormFieldHelperTextRef.current);
    }

    getClass() {
        const {className, error} = this.props;

        return classnames({
            'mdc-text-field-helper-text': true,
            'mdc-text-field-helper-text--persistent': true,
            'mdc-text-field-helper-text--validation-msg': error,
        }, className);
    }

    render() {
        const {error, helperText, style} = this.props;

        return (
            <p
                className={this.getClass()}
                aria-hidden="true"
                ref={this.textFormFieldHelperTextRef}
                style={style}
            >
                {error || helperText}
            </p>
        );
    }
}

FormFieldHelperText.propTypes = {
    /** className from parent to style from parent scss file */
    className: PropTypes.string,
    /**
        Text that will shows below the form fields and marks
        it as a validation-msg for className for additional sytling.
        If set, this will override helperText
    */
    error: PropTypes.string,
    /** text that will shows below the form fields */
    helperText: PropTypes.string,
    /** additional inline styling for the component */
    style: PropTypes.object,
};

FormFieldHelperText.defaultProps = {
    helperText: '',
};
