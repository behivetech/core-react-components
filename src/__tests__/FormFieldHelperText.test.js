import FormFieldHelperText from '../FormFieldHelperText';

describe('FormFieldHelperText', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<FormFieldHelperText helperText="mock help" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with an error set', () => {
        const wrapper = shallow(<FormFieldHelperText error="mock error" />);

        expect(wrapper).toMatchSnapshot();
    });
});
