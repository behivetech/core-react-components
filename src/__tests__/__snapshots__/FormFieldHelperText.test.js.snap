// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldHelperText should render correctly 1`] = `
<p
  aria-hidden="true"
  className="mdc-text-field-helper-text mdc-text-field-helper-text--persistent"
>
  mock help
</p>
`;

exports[`FormFieldHelperText should render with an error set 1`] = `
<p
  aria-hidden="true"
  className="mdc-text-field-helper-text mdc-text-field-helper-text--persistent mdc-text-field-helper-text--validation-msg"
>
  mock error
</p>
`;
